File: core/src/org/rexellentgames/dungeon/game/State.java
package org.rexellentgames.dungeon.game;

import box2dLight.RayHandler;
import com.badlogic.gdx.physics.box2d.World;

public class State {
	protected World world;
	protected RayHandler light;

	public State() {

	}

	public void init() {

	}

	public void destroy() {

	}

	public void update(float dt) {

	}

	public void render() {

	}

	public World getWorld() {
		return this.world;
	}

	public RayHandler getLight() {
		return this.light;
	}
}

File: BurningKnight/Game/State.cs
namespace BurningKnight.Game
{
	public class State
	{
		public Area area;
		public Area ui;
		public static InGameState InGame = new InGameState();
		
		public virtual void Init()
		{
			area = new Area();
			ui = new Area();
		}

		public virtual void Destroy()
		{
			area.Destroy();
			ui.Destroy();
		}

		public virtual void Update(float dt)
		{
			area.Update(dt);
			ui.Update(dt);
		}

		public virtual void Draw()
		{
			
		}

		public virtual void DrawUi()
		{
			
		}
	}
}